### PR TITLE
feat: claim a license

### PR DESCRIPTION
**Pull request by jrasm91**
...and then a few other things I guess.

- migrate to nestjs
- use controllers
- use services
- use interfaces
- use repositories
- have a `npm start` command that works
- use `cjs` instead of `esm`
- no broken imports missing extensions
- use the nestjs logger
- update prettier config
- auto format json files
- add new lines to end of files
- auto-sort ts imports
- manage cron jobs via the nestjs/schedule module
- use class-validation for body schema checks
- run app with tini (handles signals, doesn't phone home)
- run app as user: node, instead of root

### CODE REVIEW

File: src/interfaces/discord.interface.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  GithubStatus = '1240662502912692236',
+  Stripe = '1263492970691297300',
+}
+
+export enum DiscordEvents {
+  Ready = 'ready',
+  MessageCreate = 'messageCreate',
+  InteractionCreate = 'interactionCreate',
+  Error = 'error',
+}
```
Comment (danieldietzler) [inline src/interfaces/discord.interface.ts:18]: Why did you move those out of the constants file, but `roles` and `tags` not?

File: src/repositories/discord.repository.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+});
+
+export class DiscordRepository extends EventEmitter implements IDiscordInterface {
+  constructor() {
+    super();
+
+    bot.on('ready', () => void this.emit('ready'));
+    bot.on('error', (error) => void this.emit('error', error));
+    bot.on('interactionCreate', (interaction) => void bot.executeInteraction(interaction));
+    bot.on('messageCreate', async (message) => void bot.executeCommand(message));
```
Comment (danieldietzler) [inline src/repositories/discord.repository.ts:35]: Why did you decide to not await those?

File: src/services/discord.service.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import { DiscordChannel, IDiscordInterface } from 'src/interfaces/discord.interface';
+import { logError } from 'src/util';
+
+@Injectable()
+export class DiscordService {
+  private logger = new Logger(DiscordService.name);
+
+  constructor(@Inject(IDiscordInterface) private discord: IDiscordInterface) {
+    this.discord
+      //
```
Comment (danieldietzler) [inline src/services/discord.service.ts:14]: What's this comment doing here?

**Review by danieldietzler — COMMENTED** [review]
So this is where the bot attains it's majority isn't it? 🥹

File: src/interfaces/discord.interface.ts (hunk 1 above)
Comment (jrasm91) [reply src/interfaces/discord.interface.ts:18]: Because I needed to use it in the discord interface.

File: src/services/discord.service.ts (hunk 3 above)
Comment (jrasm91) [reply src/services/discord.service.ts:14]: Just to keep the events on multiple lines. I thought I was going to add more events

File: src/repositories/discord.repository.ts (hunk 2 above)
Comment (jrasm91) [reply src/repositories/discord.repository.ts:35]: It was throwing a type error. This might be wrong though.

File: src/interfaces/discord.interface.ts (hunk 1 above)
Comment (danieldietzler) [reply src/interfaces/discord.interface.ts:18]: You don't want to import from utils?

File: src/repositories/discord.repository.ts (hunk 2 above)
Comment (danieldietzler) [reply src/repositories/discord.repository.ts:35]: Ah I see, can look into it again next week :)

File: src/interfaces/discord.interface.ts (hunk 1 above)
Comment (jrasm91) [reply src/interfaces/discord.interface.ts:18]: I don't want to write typeof or keyof

**Review by danieldietzler — APPROVED** [review]
LGTM
